README
NAME
    Pod::Wordlist::hanekomu - Add words for spell checking POD

VERSION
    version 1.102640

SYNOPSIS
        use Test::Spelling;
        use Pod::Wordlist::hanekomu;
        all_pod_files_spelling_ok('lib');

DESCRIPTION
    This module, when loaded, adds stopwords for POD spell checking, that
    is, words that should be ignored by the spell check.

INSTALLATION
    See perlmodinstall for information and options on installing Perl
    modules.

BUGS AND LIMITATIONS
    No bugs have been reported.

    Please report any bugs or feature requests through the web interface at
    <http://rt.cpan.org>.

AVAILABILITY
    The latest version of this module is available from the Comprehensive
    Perl Archive Network (CPAN). Visit <http://www.perl.com/CPAN/> to find a
    CPAN site near you, or see
    <http://search.cpan.org/dist/Pod-Wordlist-hanekomu/>.

    The development version lives at
    <http://github.com/hanekomu/Pod-Wordlist-hanekomu> and may be cloned
    from <git://github.com/hanekomu/Pod-Wordlist-hanekomu>. Instead of
    sending patches, please fork this project using the standard git and
    github infrastructure.

AUTHOR
    Marcel Gruenauer <marcel@cpan.org>

COPYRIGHT AND LICENSE
    This software is copyright (c) 2009 by Marcel Gruenauer.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



Changes
Revision history for Perl extension Pod-Wordlist-hanekomu

1.130240  2013-01-23 20:27:57 America/New_York

1.122840  2012-10-10 10:25:10 America/New_York

    - added: iteratively

1.121370  2012-05-16 17:02:39 America/New_York

    - added: GUIDs, UUID, UUIDs, searchable

1.121231  2012-05-02 16:12:36 America/New_York

    - added deserialized, parameterizable, pre, precompute,
      precomputes, symlinked

1.120920  2012-04-01 11:53:06 Europe/Paris
    - added: prereq(s), pre, precompute, symlinked

1.120740  2012-03-14 16:36:02 EST5EDT
    - added: EINTR, chunked, redirections, timestamp, CamelCase, CPANPLUS,
      GUID, HTTPS, IETF, IRC, ISP, ISP's, JSON, modulino, SMTP, SSL, URI's,
      UTC, wiki, analyses, locator

1.113620  2011-12-28 22:25:25 Europe/Vienna
    - added "Lapworth", "UTF", "aggregator", "aggregators", "probe's",
      "runtime", "seekable" and "sigils"

1.110090  2011-01-09 12:39:53 Europe/Vienna
    - added "breakpoint", "breakpoints", "debugger's"
    - also added author names "Cushing", "Doherty", "Doran".

1.102640  2010-09-21 14:49:22 Europe/Vienna
    - added "multi", "multi-value", "multi-valued"

1.102630  2010-09-20 15:21:26 Europe/Vienna
    - added "benchmarked"

1.102340  2010-08-22 20:04:59 Europe/Vienna
    - added "Metheringham" (nigelm) to make Pod-Weaver-PluginBundle-MARCEL pass

1.102250  2010-08-13 14:33:51 Europe/Vienna
    - added: afterwards, nestable, lookup. Are these such strange words or is
      my spell checker simply a bit... retarded?
    - also added: SQL, lookups, variable's, munger, munging, storages, filenames

1.100860  2010-03-27 23:53:59 Europe/Vienna
    - converted the distribution to Dist::Zilla-style

0.02 Thu 2009.10.29 16:54:01 CET (Marcel Gruenauer <marcel@cpan.org>)
    - more words

0.01 2009-06-07T16:05:29Z (Marcel Gruenauer <marcel@cpan.org>)
    - original version


lib/Pod/Wordlist/hanekomu.pm
use 5.008;
use strict;
use warnings;

package Pod::Wordlist::hanekomu;
{
  $Pod::Wordlist::hanekomu::VERSION = '1.130240';
}
# ABSTRACT: Add words for spell checking POD
use Test::Spelling;
set_spell_cmd 'aspell list';
use utf8;
our @person_names = qw(
  Achim
  Adam
  Cushing
  Damian Conway
  Doherty
  Doran
  Eilam
  Ekker
  Florian
  Marcel Gruenauer
  Heinz
  Helmberger
  Hofstetter
  Lapworth
  Mark
  Metheringham
  Ran
  Ricardo Signes rjbs
  Takesako
  Tatsuhiko Miyagawa
  mst
);
our @tech_names = qw(
  AnyEvent
  AOP
  API
  AspectJ
  Babelfish
  CamelCase
  Coro
  CPAN
  CPANPLUS
  DateTime
  DBI
  Django
  DSL
  EINTR
  EPP
  Firefox
  FirePHP
  FIXME
  GraphViz
  GUID
  GUIDs
  HTTP
  HTTPS
  IETF
  IP
  IPv4
  IPv6
  IRC
  ISP
  ISP's
  JSON
  MakeMaker
  Markdown
  Middleware
  MongoDB
  mkdn
  modulino
  MVC
  OO
  OOP
  PARC
  PHP
  Plack
  PSGI
  RDBMS
  README
  ShipIt
  SMTP
  Spiffy
  SQL
  SQLite
  SSL
  STDERR
  STDIN
  STDOUT
  svk
  TIMTOWTDI
  Unicode
  URI
  URI's
  URIs
  UTC
  UTF
  UUID
  UUIDs
  W3CDTF
  wiki
  XS
  YAML
  YAML's
);
our @stopwords = qw(
  adaptee
  adaptees
  administrativa
  afterwards
  aggregator
  aggregators
  analyses
  array's
  backend
  behaviour
  benchmarked
  blog
  blogs
  breakpoint
  breakpoints
  bugtracker
  bundle's
  callback
  callbacks
  callee
  chomp
  chomps
  chunked
  configurator
  configurators
  crosscutting
  debugger's
  denormalized
  deserialized
  distname
  dotfile
  dotfiles
  filename
  filenames
  formatter
  github
  hash's
  homepage
  hostname
  indices
  init
  iteratively
  japanese
  Joseki
  kwalitee
  locator
  lookup
  lookups
  marshalling
  metadata
  middleware
  mixin
  monkeypatch
  monkeypatches
  monkeypatching
  multi
  multi-value
  multi-valued
  munge
  munger
  munging
  namespace
  namespaces
  nestable
  ok
  op
  parameterizable
  pipe's
  placeholders
  pluggable
  plugin's
  plugins
  pointcut
  pointcuts
  pre
  precompute
  precomputes
  prepends
  preprocessed
  prereq
  prereqs
  probe's
  redirections
  redispatch
  ref
  reusability
  runtime
  san
  searchable
  seekable
  segment's
  shipit
  sigils
  startup
  stopword
  stopwords
  storable
  storages
  stringification
  stringifications
  stringifies
  stringify
  subclass
  subclasses
  subdirectories
  subdirectory
  subobjects
  symlinked
  terminal's
  timestamp
  tokenizes
  toolchain
  tuple
  unblessed
  unshifts
  username
  uuid
  value's
  variable's
  vim
  wellformedness
  whitelist
  whitelists
  workflow
  workflows
  wormhole
  yml
);

# dedup
my %words = map { $_ => 1 } @person_names, @tech_names, @stopwords;
add_stopwords(sort keys %words);
$ENV{LANG} = 'C';
1;

__END__

=pod

=head1 NAME

Pod::Wordlist::hanekomu - Add words for spell checking POD

=head1 VERSION

version 1.130240

=head1 SYNOPSIS

    use Test::Spelling;
    use Pod::Wordlist::hanekomu;
    all_pod_files_spelling_ok('lib');

=head1 DESCRIPTION

When loaded, this module automatically adds Test::Spelling stopwords for POD
spell checking, that is, words that should be ignored by the spell check.

The stopword list includes some CPAN author names, technical terms (e.g.  JSON,
URI) and other commonly-used words not included in the default word list (e.g.
mixin, munging).

=head1 INSTALLATION

See perlmodinstall for information and options on installing Perl modules.

=head1 BUGS AND LIMITATIONS

You can make new bug reports, and view existing ones, through the
web interface at L<http://rt.cpan.org/Public/Dist/Display.html?Name=Pod-Wordlist-hanekomu>.

=head1 AVAILABILITY

The latest version of this module is available from the Comprehensive Perl
Archive Network (CPAN). Visit L<http://www.perl.com/CPAN/> to find a CPAN
site near you, or see L<https://metacpan.org/module/Pod::Wordlist::hanekomu/>.

=head1 AUTHORS

=over 4

=item *

Marcel Gruenauer <marcel@cpan.org>

=item *

David Golden <dagolden@cpan.org>

=back

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2009 by Marcel Gruenauer.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut


Makefile.PL

use strict;
use warnings;

use 5.008;

use ExtUtils::MakeMaker 6.30;



my %WriteMakefileArgs = (
  "ABSTRACT" => "Add words for spell checking POD",
  "AUTHOR" => "Marcel Gruenauer <marcel\@cpan.org>, David Golden <dagolden\@cpan.org>",
  "BUILD_REQUIRES" => {
    "Carp" => 0,
    "File::Find" => 0,
    "File::Temp" => 0,
    "Scalar::Util" => 0,
    "Test::More" => "0.94"
  },
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.30"
  },
  "DISTNAME" => "Pod-Wordlist-hanekomu",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "NAME" => "Pod::Wordlist::hanekomu",
  "PREREQ_PM" => {
    "Test::Spelling" => 0,
    "strict" => 0,
    "utf8" => 0,
    "warnings" => 0
  },
  "VERSION" => "1.130240",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.56) } ) {
  my $br = delete $WriteMakefileArgs{BUILD_REQUIRES};
  my $pp = $WriteMakefileArgs{PREREQ_PM};
  for my $mod ( keys %$br ) {
    if ( exists $pp->{$mod} ) {
      $pp->{$mod} = $br->{$mod} if $br->{$mod} > $pp->{$mod};
    }
    else {
      $pp->{$mod} = $br->{$mod};
    }
  }
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



